23.js

```diff
@@ -0,0 +1,159 @@
+// You cannot pour more gas then gasTank
+// Create method printInfo that prints in console all available information
+
+// Create class Car
+// Car should inherit from Vehicle
+// Initialize Car with additional properties type and maxPassengerCount
+// Examples of type SEDAN, MINIVAN, SPORTS CAR...
+// Update method printInfo that prints in console all available information
+
+// Create class Bus
+// Bus should inherit from Car
+// Create method updatePassengers that receives argument passengers and updates passengerCount to that number
+// passengerCount cannot be more then maxPassengerCount and less then 0
+// Update method printInfo that prints in console all available information
+
+class Vehicle {
+  constructor(power, gasTank, mass) {
+    this.power = power;
+    this.gasTank = gasTank;
+    this.mass = mass;
+    this.maxSpeed = this.maxSpeedFormula();
+    this.gasUsage = this.gasUsageFormula();
+  }
+  magicNumber = 0.84;
+  started = false;
+  speed = 0;
+  distance = 0;
+
+  maxSpeedFormula = () => {
+    return (this.magicNumber * this.power) / this.mass;
+  };
+  gasUsageFormula = () => {
+    return this.maxSpeed / this.power;
+  };
+
+  getMaxSpeed = () => {
+    return this.maxSpeed;
+  };
+  getGasUsage = () => {
+    return this.gasUsage;
+  };
+  startEngine = () => {
+    if (this.gasTank > 0) {
+      this.started = true;
+    }
+    return this;
+  };
+
+  stopEngine = () => {
+    this.started = false;
+    return this;
+  };
+  drive = (speed, distance) => {
+    if (this.started === false) {
+      console.log("Please, start engine");
+    }
+    if (speed > this.maxSpeed && speed <= 0) {
+      console.log("Please, check your speed");
+    }
+    const spent = distance * this.gasUsage; // gas, that we spend on 1 km
+    if (spent >= this.gasTank) {
+      console.log("Please, add gas");
+    } else {
+      this.updateGas(spent);
+      this.speed = speed;
+      this.distance = distance;
+    }
+    return this;
+  };
+
+  updateGas = (number) => {
+    this.gasTank = this.gasTank - number;
+    return this.gasTank;
+  };
+
+  addGas = (number) => {
+    if (number > 0) {
+      this.gasTank = number + this.gasTank;
+    }
+
+    return this.gasTank;
+  };
+
+  printInfo() {
+    const info = {
+      power: this.power,
+      gasTank: this.gasTank,
+      speed: this.speed,
+      distance: this.distance,
+      isStarted: this.started,
+    };
+    return info;
+  }
+}
+const myVehicle = new Vehicle(220, 40, 2);
+myVehicle.startEngine().drive(30, 30).addGas(5);
+
+class Car extends Vehicle {
+  constructor(type, maxPassengerCount, ...rest) {
+    super(...rest);
+    this.type = type;
+    this.maxPassengerCount = maxPassengerCount;
+  }
+  printInfo() {
+    const info = {
+      power: this.power,
+      gasTank: this.gasTank,
+      speed: this.speed,
+      distance: this.distance,
+      isStarted: this.started,
+      type: this.type,
+      maxPassengerCount: this.maxPassengerCount,
+    };
+    return info;
```

Comment:
you don't need to build this obj from scratch 
you can call super.printInfo() and then extend obj with new properties